app_dart/lib/src/request_handlers/get_build_status.dart

```diff
@@ -40,4 +55,52 @@ base class GetBuildStatus extends RequestHandler<Body> {
       failingTasks: status.failedTasks,
     );
   }
+
+  /// Sends tree status updates to discord when they change from what was
+  /// last sent.
+  Future<void> maybeNotifyDiscord(BuildStatus status) async {
```

Comment:
Can you move the bulk of this into a `DiscordService` (or just `Discord`)?

Ideally that class would not know anything about the rest of Cocoon (I.e. not import or use Firestore, or anything like that). In fact I'd imagine the public API is limited to:

```dart
interface class Discord {
  Future<DiscordResponse> postTreeStatusMessage(String message);
}
```

Reply:
Sure, that part is easy.